buf/registry/module/v1beta1/commit_service.proto

```diff
@@ -310,14 +310,28 @@ message GetBlobsResponse {
 }
 
 message GetMissingBlobDigestsRequest {
-  // The digests to see if we have Blobs for.
-  repeated buf.registry.storage.v1beta1.Digest digests = 1 [
+  message ModuleBlobDigests {
+    // The reference to the Module that the digests belong to.
+    ModuleRef module_ref = 1 [(buf.validate.field).required = true];
+    // The digests to see if we have Blobs for.
+    repeated buf.registry.storage.v1beta1.Digest blob_digests = 2 [
+      (buf.validate.field).repeated.min_items = 1,
+      (buf.validate.field).repeated.max_items = 250
+    ];
+  }
+  repeated ModuleBlobDigests module_blob_digests = 1 [
     (buf.validate.field).repeated.min_items = 1,
     (buf.validate.field).repeated.max_items = 250
   ];
 }
 
 message GetMissingBlobDigestsResponse {
+  message MissingBlobDigest {
```

Comment:
This isn't a single digest, this is a set of Digests, so this name doesn't make as much sense. I would want to call it `ModuleDigests`, but then the field `repeated ModuleDigests module_digests` doesn't really make sense, it's more like `repeated ModuleDigests module_digests_es`. Plurals for message names rarely make sense.

Perhaps something like `ModuleBlobDigestSet`, and then `repeated ModuleBlobDigestSet module_blob_digest_sets`, or just `Value` and `repeated Value values`.

Reply:
Done